pkg/kubelet/cm/topologymanager/topology_manager.go

```diff
@@ -202,26 +202,37 @@ func (m *manager) RemoveContainer(containerID string) error {
 }
 
 func (m *manager) Admit(attrs *lifecycle.PodAdmitAttributes) lifecycle.PodAdmitResult {
-	klog.Infof("[topologymanager] Topology Admit Handler")
+	// Short circuit to unconditionally admit the pod if the policy is 'none'.
 	if m.policy.Name() == "none" {
-		klog.Infof("[topologymanager] Skipping calculate topology affinity as policy: none")
 		return lifecycle.PodAdmitResult{
 			Admit: true,
 		}
 	}
+
 	pod := attrs.Pod
-	c := make(map[string]TopologyHint)
 
+	// Short circuit to unconditionally admit the pod if it is already running.
+	// This can happen after a kubelet restart, for example.
+	if pod.Status.Phase == v1.PodRunning && pod.DeletionTimestamp == nil {
+		klog.Infof("[topologymanager] Skipping ToplogyHint generation since pod already running")
```

Comment:
maybe add pod information to this log so we know what pod it applies to?

Reply:
at this stage it is guaranteed that all resources have been assigned to the pod right?

Reply:
Yes. If it is already running, then there will be no more resources handed out to it.